utils/navigation_decorator.py

```diff
@@ -0,0 +1,12 @@
+import cfme.fixtures.pytest_selenium as sel
+import cfme.web_ui.menu as menu
+from method_decorator import method_decorator
+
+
+def navigate(page_name):
+    class Decorator(method_decorator):
+        def __call__(self, *args, **kwargs):
+            if menu.get_current_page_name() != page_name:
```

Comment:
I don't think this is gonna help as objects extremely rarely live in a menu item.